internal/cas/fetch.go

```diff
@@ -69,3 +69,31 @@ func FetchManifest(ctx context.Context, target oras.Target, reference string, p
 	defer rc.Close()
 	return content.ReadAll(rc, desc)
 }
+
+// FetchBlob fetches the blob content of reference from blob store.
+func FetchBlob(ctx context.Context, blob oras.Target, reference string) ([]byte, error) {
+	rf := blob.(registry.ReferenceFetcher)
+	desc, rc, err := rf.FetchReference(ctx, reference)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+	return content.ReadAll(rc, desc)
+}
+
+// BlobTarget returns a ORAS Target with a no-op Tag method wrapping the
+// provided blob store b.
+func BlobTarget(b registry.BlobStore) oras.Target {
```

Comment:
You may leverage [`oras.ReadOnlyTarget`](https://github.com/oras-project/oras-go/blob/main/target.go#L34).